### PR TITLE
[Spot] Avoid spot jobs files being overwritten on spot controller

### PR DESCRIPTION
**Pull request by Michaelvll**
To reproduce the overwriting problem:
1. `sky spot launch -n test 'echo hi'` 
2. `sky spot launch -n test 'echo hi2'` in parallel with the first one.
One of the job will overwrite the test.yaml and test.config_yaml of the other one on the spot controller, which may cause an issue that the two tasks have the same run commands, if we add a `sleep 40` before the controller process starts in `spot-controller.yaml.j2`, i.e. both task will show `hi2`.

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - the snippets above
  - [x] `sky launch --cloud gcp --gpus tpu-v2-8`
  - [x] `sky launch --cloud gcp --cpus 2+`
- [x] All smoke tests: `pytest tests/test_smoke.py --managed-spot`

### CODE REVIEW

File: sky/execution.py [hunk 1]
```diff
@@ -630,10 +639,12 @@ def spot_launch(
...
-        yaml_path = backend_utils.fill_template(
-            spot.SPOT_CONTROLLER_TEMPLATE,
-            vars_to_fill,
-            output_prefix=spot.SPOT_CONTROLLER_YAML_PREFIX)
+        yaml_path = os.path.expanduser(
+            os.path.join(spot.SPOT_CONTROLLER_YAML_PREFIX,
+                         f'{name}-{task_uuid}.yaml'))
+        backend_utils.fill_template(spot.SPOT_CONTROLLER_TEMPLATE,
+                                    vars_to_fill,
+                                    output_path=yaml_path)
```
Comment (concretevitamin) [inline sky/execution.py:647]: output_prefix arg can be removed now, it seems

File: sky/execution.py [hunk 2]
```diff
@@ -570,6 +570,14 @@ def spot_launch(
 
     task = _maybe_translate_local_file_mounts_and_sync_up(task)
 
+    if name is None:
+        if task.name is not None:
+            name = task.name
+        name = backend_utils.generate_cluster_name()
```
Comment (concretevitamin) [inline sky/execution.py:576]: add `else:`, otherwise overrides L575?

File: sky/execution.py [hunk 3]
```diff
@@ -581,7 +589,8 @@ def spot_launch(
             'user_yaml_path': f.name,
             'user_config_path': None,
             'spot_controller': controller_name,
-            'cluster_name': name,
+            'task_name': name,
```
Comment (concretevitamin) [inline sky/execution.py:592]:
```suggestion
            'task_name': name,  # actual cluster name will be <task_name>-<spot job ID>
```

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks for identifying this issue.

File: sky/execution.py (hunk 3 above)
Comment (Michaelvll) [reply sky/execution.py:594]: Done. Thanks!

File: sky/execution.py (hunk 1 above)
Comment (Michaelvll) [reply sky/execution.py:648]: Good point! Removed. Thanks!

File: sky/execution.py (hunk 2 above)
Comment (Michaelvll) [reply sky/execution.py]: Oops, good catch! Fixed. Thanks!

**Pull request merged**